src/react-utils/useProxy.ts

```diff
@@ -20,18 +20,20 @@ import { useLayoutEffect } from 'react'
  * @param proxy
  * @returns A new proxy which you can use in the render as well as in callbacks.
  */
-export const useProxy = <T extends object>(proxy: T) => {
-  const snapshot = useSnapshot(proxy)
+export function useProxy<T extends object>(proxy: T): T {
+  const snapshot = useSnapshot(proxy) as T
 
   let isRendering = true
 
   useLayoutEffect(() => {
+    // This is an intentional hack
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Does the linter really error on this? Not passing the dep array is legit usage.

Reply:
Yes, it does. What it says is useRef instead. It's legit, but not best practice in general?

Reply:
Ah yeah, I see the warning, sorry. It's funny. I couldn't help figuring out a linter compliant version:

```tsx
const isRendering = { val: true }

useLayoutEffect(() => {
  isRendering.val = false
})
```

Now that you can call a hack 😜

Reply:
Not worth increasing the bundle size. It's a hack anyway for either case, from the react perspective.

Reply:
Of course, I was just kidding. Thanks @dai-shi for publishing 🥳 and all the great work!